services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/PortPropertiesSpec.groovy

```diff
@@ -0,0 +1,251 @@
+            it.portNumber == nonExistentPort
+            it.discoveryEnabled == DISCOVERY_ENABLED_DEFAULT
+        }
+
+        when: "Try to update port discovery property for non-existing port"
+        northboundV2.updatePortProperties(sw.dpId, nonExistentPort, new PortPropertiesDto(discoveryEnabled: true))
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.statusCode == HttpStatus.NOT_FOUND
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Port not found: 'Port FSM not found (${sw.dpId}_${nonExistentPort}).'"
+    }
+
+    @Tags(VIRTUAL)
+    def "System doesn't discover link when port discovery property is disabled"() {
+        given: "A deleted link"
+        def sw = topology.activeSwitches.first()
+        def relatedIsls = topology.getRelatedIsls(sw)
+        def islToManipulate = relatedIsls.first()
+
+        // Bring port down on the src switch
+        antiflap.portDown(islToManipulate.srcSwitch.dpId, islToManipulate.srcPort)
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            islUtils.getIslInfo(islToManipulate).get().state == IslChangeType.FAILED
+        }
+
+        // delete link
+        northbound.deleteLink(islUtils.toLinkParameters(islToManipulate))
+
+        when: "Disable port discovery property on the src and dst switches"
+        northboundV2.updatePortProperties(islToManipulate.srcSwitch.dpId, islToManipulate.srcPort,
+                new PortPropertiesDto(discoveryEnabled: false))
+        northboundV2.updatePortProperties(islToManipulate.dstSwitch.dpId, islToManipulate.dstPort,
+                new PortPropertiesDto(discoveryEnabled: false))
+
+        and: "Bring port up on the src switch"
+        antiflap.portUp(islToManipulate.srcSwitch.dpId, islToManipulate.srcPort)
+
+        then: "Link is not detected"
+        Wrappers.timedLoop(discoveryInterval + WAIT_OFFSET / 2) {
+            assert !islUtils.getIslInfo(islToManipulate).isPresent()
+        }
+
+        when: "Deactivate/activate src switch"
+        lockKeeper.knockoutSwitch(sw)
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            assert northbound.getSwitch(sw.dpId).state == SwitchChangeType.DEACTIVATED
+            assert northbound.getAllLinks().findAll {
+                it.state == IslChangeType.FAILED
+            }.size() == (relatedIsls - islToManipulate).size() * 2
+        }
+
+        lockKeeper.reviveSwitch(sw)
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            assert northbound.getSwitch(sw.dpId).state == SwitchChangeType.ACTIVATED
+            def links = northbound.getAllLinks()
+            (relatedIsls - islToManipulate).each {
+                assert islUtils.getIslInfo(links, it).get().state == IslChangeType.DISCOVERED
+            }
+        }
+
+        then: "Link is still not detected"
+        Wrappers.timedLoop(discoveryInterval + WAIT_OFFSET / 2) {
```

Comment:
I just realized that actually we can reduce the time of this timedLoop. Since all other switch-related isls are already discovered (L161) we can actually assume that at least `discoveryInterval` time has passed since the switch was activated, can we?

Reply:
good idea,
done